x-pack/solutions/chat/plugins/workchat-app/public/application/components/integrations/listing/integration_catalog_view.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiCard,
+  EuiFlexGrid,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiIcon,
+  EuiSpacer,
+  EuiText,
+  IconType,
+  useEuiTheme,
+} from '@elastic/eui';
+import { IntegrationType } from '@kbn/wci-common';
+import { KibanaPageTemplate } from '@kbn/shared-ux-page-kibana-template';
+import { i18n } from '@kbn/i18n';
+import { IntegrationListView } from './integration_list_view';
+import { getIntegrationIcon } from '../utils';
+import { useNavigation } from '../../../hooks/use_navigation';
+import { appPaths } from '../../../app_paths';
+
+interface IntegrationCardData {
+  title: string;
+  icon: IconType;
+  description: string;
+  disabled?: boolean;
+}
+
+export const IntegrationCatalogView: React.FC = () => {
+  const { navigateToWorkchatUrl } = useNavigation();
+
+  const integrationCards: Record<IntegrationType, IntegrationCardData> = {
```

Comment:
This gets re-evaluated at each rendering of the component. Move outside of the component (you could also use `useMemo`, but moving outside here is strictly better given you don't depend on any of the component's props).